sap/rfc/user.py

```diff
@@ -62,6 +63,61 @@ def today_sap_date() -> str:
     return sap_date_from(datetime.date.today())
 
 
+class UserPasswordManager:
+    """An utility class for handling productive password"""
+
+    def __init__(self):
+        self._use_productive_password = None
+        self._password = None
+        self._user_type = None
+
+    def get_use_productive_password(self) -> bool:
+        """Get if new password should be productive or not"""
+
+        return self._use_productive_password
+
+    def get_password(self) -> str:
+        """Get user's new password"""
+
+        return self._password
+
+    def get_user_type(self) -> str:
+        """Get user type"""
+
+        return self._user_type
+
+    def get_dummy_password(self) -> str:
+        """Get dummy password"""
+
+        return os.environ.get('SAPCLI_ABAP_USER_DUMMY_PASSWORD', 'DummyPwd123!')
+
+    def set_use_productive_password(self, use_productive_password: bool):
+        """Set if new password should be productive or not"""
+
+        self._use_productive_password = use_productive_password
+        return self._use_productive_password
+
+    def set_password(self, password):
+        """Set user's new password"""
+
+        self._password = password
+        return self._password
+
+    def set_user_type(self, user_type: str):
+        """Set user type"""
+
+        self._user_type = user_type
+        return self._user_type
+
+    def is_productive_password_requested(self) -> bool:
+        """Is it requested to use a productive password for Dialog or Communications Data user"""
+
+        return all([
+            self.get_use_productive_password(),
+            self.get_password(),
+            self.get_user_type() in ['A', 'C']])     # A = Dialog, C = Communications Data
```

Comment:
I could not find unit tests checking all the possible combinations.

Reply:
Do i get it right that "is_productive_password_requested" returns FALSE even if a developer set "productive" password but user type is not "A" or "C"? I think we should be more verbose and made consumers know that productive password cannot be set.

Reply:
The method name was changed to `def is_productive_password_needed()`. Hope it clear now. Also I have added unit tests.